src/insert-text-at-cursor.js

```diff
@@ -86,7 +86,12 @@ export default function (input, text) {
 
           // Finally insert a new node. The browser will automatically
           // split start and end nodes into two if necessary
-          range.insertNode(textNode)
+          if (range.commonAncestorContainer.nodeName === '#text') {
```

Comment:
It'd be useful to have a good solid comment that explains how this works since it's super hard to understand.

Reply:
Added comment trying to explain why are we testing the `nodeName`